### PR TITLE
Add ImHex

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/WerWolv/ImHex/issues/511
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/WerWolv/ImHex/pull/70/ https://github.com/WerWolv/ImHex/pull/513

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build net.werwolv.ImHex

**Comment by Mailaender** [comment]
bot, build net.werwolv.ImHex

File: net.werwolv.ImHex.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>net.werwolv.ImHex.desktop</id>
```
Comment (hfiguiere) [inline net.werwolv.ImHex.appdata.xml:3]:
```suggestion
  <id>net.werwolv.ImHex</id>
```

File: net.werwolv.ImHex.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+rename-desktop-file: imhex.desktop
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --filesystem=home
+  - --device=dri
+
+modules:
+  - name: libiconv
```
Comment (hfiguiere) [inline net.werwolv.ImHex.yaml:15]: are you sure you need libiconv? It has been part of glibc for a long time.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by Mailaender** [comment]
bot, build net.werwolv.ImHex

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: :-(

Comment (Mailaender) [reply flathub.json:2]: https://flathub.org/builds/#/builders/5/builds/7379

Comment (hfiguiere) [reply flathub.json:2]: yes I saw. and that still make me :-(

Comment (tinywrkb) [reply flathub.json:2]: Maybe try building without the git submodules, and build the dependencies separately. The AUR package has [this patch](https://aur.archlinux.org/cgit/aur.git/tree/0001-makepkg-Remove-external-stuff.patch?h=imhex).

File: net.werwolv.ImHex.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      - -DGLFW_BUILD_DOCS=OFF
+    sources:
+      - type: archive
+        url: https://github.com/glfw/glfw/releases/download/3.3.2/glfw-3.3.2.zip
+        sha256: 08a33a512f29d7dbf78eab39bd7858576adcc95228c9efe8e4bc5f0f3261efc7
+    cleanup:
+      - /include
+      - /lib/pkgconfig
+
+  - name: mbedtls
```
Comment (barthalion) [inline net.werwolv.ImHex.yaml:31]: There are a bunch of unrelated binaries in `/app/bin` that probably come from this. They should be cleaned up.

**Comment by Mailaender** [comment]
bot, build net.werwolv.ImHex

**Comment by Mailaender** [comment]
I can't get this to run anymore.

**Comment by barthalion** [comment]
Works just fine for me.

File: flathub.json (hunk 3 above)
Comment (Mailaender) [reply flathub.json]: It seems building FMT separately solved this.

**Comment by barthalion** [comment]
```
[📦 net.werwolv.ImHex ~]$ grep Exec= /app/share/applications/net.werwolv.ImHex.desktop 
Exec=/usr/bin/imhex %U
```

There you go. It launches only from CLI, not from the desktop file.

**Comment by Mailaender** [comment]
bot, build net.werwolv.ImHex

File: desktop.patch [hunk 5]
```diff
@@ -0,0 +1,46 @@
+From 93af955a2976330cbdb855f1491500653c534152 Mon Sep 17 00:00:00 2001
```
Comment (hfiguiere) [inline desktop.patch:1]: you could also, instead of a patch, use `desktop-file-edit`.
not blocking anything though.

**Pull request closed without merging**